src/components/InventoryTable/hooks/useColumns.js

```diff
@@ -16,13 +18,13 @@ const useColumns = (columnsProp, disableDefaultColumns, showTags, columnsCounter
     );
     const disabledColumns = Array.isArray(disableDefaultColumns) ? disableDefaultColumns : [];
     const defaultColumnsFiltered = useMemo(() => (disableDefaultColumns === true) ?
-        [] : defaultColumns().filter(({ key }) =>
+        [] : defaultColumns(groupsEnabled).filter(({ key }) =>
             isColumnEnabled(key, disabledColumns, showTags)
         ), [disabledColumns, disableDefaultColumns, showTags]);
```

Comment:
We need to add `groupsEnabled` to these deps array so that the value is updated when the flag changes.

Reply:
Ok